Documents/Documents/Code/Business/Providers/ASCiCloudProvider.swift

```diff
@@ -712,38 +717,33 @@ class ASCiCloudProvider: ASCFileProviderProtocol & ASCSortableFileProviderProtoc
                         completeon?(strongSelf, nil, false, ASCProviderError(msg: error.localizedDescription))
                     } else {
                         self?.attributesOfItem(path: remotePath, completionHandler: { [weak self] fileObject, error in
-                            DispatchQueue.main.async { [weak self] in
-                                guard let strongSelf = self else { return }
-
-                                if let error = error {
-                                    completeon?(strongSelf, nil, false, ASCProviderError(msg: error.localizedDescription))
-                                } else if let fileObject = fileObject {
-                                    let fileSize: UInt64 = (fileObject.size < 0) ? 0 : UInt64(fileObject.size)
-                                    let cloudFile = ASCFile()
-                                    cloudFile.id = fileObject.path
-                                    cloudFile.rootFolderType = .icloudAll
-                                    cloudFile.title = fileObject.name
-                                    cloudFile.created = fileObject.creationDate ?? fileObject.modifiedDate
-                                    cloudFile.updated = fileObject.modifiedDate
-                                    cloudFile.createdBy = strongSelf.user
-                                    cloudFile.updatedBy = strongSelf.user
-                                    cloudFile.parent = folder
-                                    cloudFile.viewUrl = fileObject.path
-                                    cloudFile.displayContentLength = String.fileSizeToString(with: fileSize)
-                                    cloudFile.pureContentLength = Int(fileSize)
-
-                                    ASCAnalytics.logEvent(ASCConstants.Analytics.Event.createEntity, parameters: [
-                                        ASCAnalytics.Event.Key.portal: "icloud",
-                                        ASCAnalytics.Event.Key.onDevice: false,
-                                        ASCAnalytics.Event.Key.type: ASCAnalytics.Event.Value.file,
-                                        ASCAnalytics.Event.Key.fileExt: cloudFile.title.fileExtension().lowercased(),
-                                    ])
-
-                                    completeon?(strongSelf, cloudFile, true, nil)
-                                } else {
-                                    completeon?(strongSelf, nil, false, nil)
-                                }
+                            guard let strongSelf = self else { return }
```

Comment:
Need to wrap DispatchQueue.main.async. Method attributesOfItem has async completionHandler for same providers